guides/Building Your Bot/CreatingACustomMemberGateway.md

```diff
@@ -0,0 +1,171 @@
+	}
+
+};
+
+```
+
+Alright! Now you have a fully functioning member-gateway. Now, we can create a per guild level system.
+
+## Level Up!
+
+We will be taking code from {@tutorial CreatingPointsSystems}, and modifying it a small bit to work with the newly created member-gateway.
+
+
+Create a new monitor, and add the follwing code:
+
+```javascript
+const { Monitor } = require('klasa');
+
+module.exports = class extends Monitor {
+
+	// Constructor
+
+	async run(message) {
+		// If the message was not sent in a TextChannel, ignore it.
+		if (!message.guild) return;
+
+		// Calculate the next value for experience.
+		const nextValue = message.member.configs.experience + 1;
+
+		// Cache the current level.
+		const currLevel = message.member.configs.level;
+
+		// Calculate the next level.
+		const nextLevel = Math.floor(0.1 * Math.sqrt(nextValue + 1));
+
+		// Update the members's configuration entry by adding 1 to it, and update the level also.
+		await message.member.configs.update(['experience', 'level'], [nextValue, nextLevel]);
+
+		// If the current level and the next level are not the same, then it has increased, and you can send the message.
+		if (currLevel !== nextLevel) {
+			// Send the message to the channel congratulating the user.
+			await message.send(`Congratulations! You leveled up to level **${currLevel}**!`);
+		}
+	}
+
+	// Init
+
+}
+```
+
+
+## Creating A Command To Check Level
+
+If the member wants to check there own level or experience, we can do so with two simple commands:
+
+### Points Command
+
+```javascript
+const { Command } = require('klasa');
+
+module.exports = class extends Command {
+
+	constructor(...args) {
+		super(...args, { description: 'Check how many points you have in this guild.' });
+	}
+
+	async run(message) {
+		return message.send(`You have a total of ${message.member.configs.experience} experience points!`);
+	}
+
+}
+
+```
+
+### Level Command
+
+
+```javascript
+const { Command } = require('klasa');
+
+module.exports = class extends Command {
+
+	constructor(...args) {
+		super(...args, { description: 'Check your current level in this guild.' });
+	};
+
+	async run(message) {
+		return message.send(`You are currently level ${message.member.configs.level}!`);
+	};
+
+};
+
+```
+
+## Preserved Configs
+
+Klasa has a {@link KlasaClientOptions} for preserving guild configs, incase your bot leaves a guild. This however, would not be reflected in the current member-gateway, although it is easily fixable with a guildRemove event.
+
+>Beware! This will delete all member configuration entries if your bot is kicked, you accidently leave, etc.
+
+
+```javascript
+
+const { Event } = require('klasa');
+
+module.exports = class extends Event {
+	constructor(...args) {
+		super(...args, { name: 'guildRemove' });
+	}
+	
+	async run(guild) {
+
+		// just incase of an outage, check if the guild is avialable, and also check if we are preserving configs.

```

Comment:
`in case`, `available` not `avialable`
Should be `Check if the guild is not available, or if we set the preserveConfigs option in our client` imo